numba/cuda/cudadrv/driver.py

```diff
@@ -1930,7 +1930,7 @@ def resolver(future, status):
             if future.done():
                 return
             elif status == 0:
-                future.set_result(None)
+                future.set_result(self)
```

Comment:
Is there any chance that the API might need to return something more than just the stream in the future, if so should this return a container of some sort of which `stream` is a member?

Reply:
As discussed OOB, I think I'd prefer to keep the interface simple for now and just return the stream. If extra info is needed in future, it could be enabled by an optional default-`False` kwarg on the `async_done` method.

Reply:
Thanks for recording this outcome.